src/chi-vue/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,338 @@
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { uuid4 } from '../../utils/utils';
+import {
+  activeClass,
+  inverseClass,
+  buttonClass,
+  buttonContentClass,
+  buttonGroupClass,
+  iconButtonClass,
+  lightClass,
+  buttonFlatClass,
+  disabledClass,
+  iconClass,
+  paginationClass,
+  paginationResultsClass,
+  paginationCompactClass,
+  paginationContentClass,
+  paginationLabelClass,
+  paginationPageSizeCLass,
+  paginationStartClass,
+  paginationCenterClass,
+  inputClass,
+  paginationJumperClass,
+  paginationEndClass
+} from '../../constants/classes';
+import { pageChangeEvent, pageSizeChangeEvent } from '../../constants/events';
+
+@Component
+export default class Pagination extends Vue {
+  @Prop() pages!: number;
+  @Prop() firstLast!: boolean;
+  @Prop() inverse!: boolean;
+  @Prop() currentPage!: number;
+  @Prop() results!: number;
+  @Prop() pageSize!: number;
+  @Prop() pageJumper!: number;
+  @Prop() compact!: boolean;
+  @Prop() ariaLabel!: boolean;
+  @Prop() size!: string;
+
+  _pageJumperUuid!: string;
+  _distanceFromCurrent!: number;
+  _lastRenderedPage!: number;
+  _pagesToRender: any = [];
+  _startPage!: JSX.Element | null;
+  _lastPage!: JSX.Element | null;
+  _results: any;
+  _pageSize: any;
+  goToPage: any;
+
+  constructor() {
+    super();
+  }
+
+  calculateDistanceFromCurrent() {
+    switch (this.currentPage) {
+      case 1:
+        this._distanceFromCurrent = 4;
+        break;
+      case 2:
+        this._distanceFromCurrent = 3;
+        break;
+      default:
+        this._distanceFromCurrent = 2;
+    }
+  }
+
+  _goToPage(pageChange: HTMLElement) {
+    let pageToGo;
+    for (
+      let cur = pageChange;
+      cur && !cur.classList.contains(buttonGroupClass);
+      cur = cur.parentNode as HTMLElement
+    ) {
+      if (cur.nodeName === 'BUTTON') {
+        pageToGo = cur.getAttribute('data-page');
```

Comment:
Data attributes have a specific way of getting information from them that might be preferred. Please consider: https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes